core/src/sumeragi/mod.rs

```diff
@@ -419,8 +423,8 @@ impl VotingBlock {
 /// Arguments for [`SumeragiHandle::start`] function
 #[allow(missing_docs)]
 pub struct SumeragiStartArgs {
-    pub chain_id: ChainId,
-    pub configuration: Box<Configuration>,
+    pub sumeragi_config: SumeragiConfig,
+    pub common_config: CommonConfig,
```

Comment:
I had an issue:

- there are "root level" config values like chain id, key pair, and address
- these parameters have a computed value `peer_id()`
- sumeragi needs them in order to start
- i don't want to pass the whole root config to sumeragi
- i can put these parameters to `SumeragiConfig`, but they don't strictly belong to "sumeragi" domain, and I want to avoid duplication.


The solution I came up with is to separate "common", "base" parameters into a separate struct to pass what is needed into sumeragi.

Actually, now I think it's better to simply put all of "common" parameters directly into `SumeragiConfig`.

Reply:
I tried to do it again and remembered that `Common` is really used not only in Sumeragi, and when I try to put all stuff into `SumeragiConfig`, many places become messy and too coupled.